misc/compare_pushes.py

```diff
@@ -1,116 +1,73 @@
-
-import slugid
 
 from deepdiff import DeepDiff
 from thclient import TreeherderClient
 
 logging.basicConfig()
-logging.getLogger().setLevel(logging.DEBUG)
 logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
 HOSTS = {
     "localhost": "http://localhost:8000",
     "stage": "https://treeherder.allizom.org",
     "production": "https://treeherder.mozilla.org"
 }
 
 
-def remove_some_attributes(job, production_job):
-    # I belive these differences are expected since they are dependant to when the data
-    # was inserted inside of the database
-    del job["build_platform_id"]
-    del job["id"]
-    del job["job_group_id"]
-    del job["job_type_id"]
-    del job["last_modified"]
-    del job["push_id"]
-    del job["result_set_id"]
-    del production_job["build_platform_id"]
-    del production_job["id"]
-    del production_job["job_group_id"]
-    del production_job["job_type_id"]
-    del production_job["last_modified"]
-    del production_job["push_id"]
-    del production_job["result_set_id"]
-
-    if job.get("end_timestamp"):
-        del job["end_timestamp"]
-        del job["start_timestamp"]
-        del production_job["end_timestamp"]
-        del production_job["start_timestamp"]
-
-    if job.get("failure_classification_id"):
-        del job["failure_classification_id"]
-        del production_job["failure_classification_id"]
-
-
-def print_url_to_taskcluster(job_guid):
-    job_guid = job["job_guid"]
-    (decoded_task_id, _) = job_guid.split("/")
-    # As of slugid v2, slugid.encode() returns a string not bytestring under Python 3.
-    taskId = slugid.encode(uuid.UUID(decoded_task_id))
-    logger.info("https://taskcluster-ui.herokuapp.com/tasks/%s", taskId)
-
-
-if __name__ == "__main__":
+def main(args):
+    compare_to_client = TreeherderClient(server_url=HOSTS[args.host])
+    production_client = TreeherderClient(server_url=HOSTS["production"])
+
+    # Support comma separated projects
+    projects = args.projects.split(',')
+    for _project in projects:
+        print("Comparing {} against production.".format(_project))
+        # Remove properties that are irrelevant for the comparison
+        pushes = compare_to_client.get_pushes(_project, count=50)
+        for _push in sorted(pushes, key=lambda push: push["revision"]):
+            del _push["id"]
+            for _rev in _push["revisions"]:
+                del _rev["result_set_id"]
+
+        production_pushes = production_client.get_pushes(_project, count=50)
+        for _push in sorted(production_pushes, key=lambda push: push["revision"]):
+            del _push["id"]
+            for _rev in _push["revisions"]:
+                del _rev["result_set_id"]
+
+        for index in range(0, len(pushes)):
+            assert pushes[index]["revision"] == production_pushes[index]["revision"]
+            difference = DeepDiff(pushes[index], production_pushes[index])
+            if difference:
+                print(difference.to_json())
```

Comment:
I think historically we've used logging.info to print to console rather than using `print`. Is there a reason why it'd be better to do it this way? :)

Reply:
I was being lazy. I saw `INFO:__main__:` in the messages and did not want to configure the logger to get rid of them. I've switched it back.